libcloud/compute/drivers/ovh.py

```diff
@@ -113,29 +118,31 @@ def create_node(self, name, image, size, location, ex_keyname=None):
         :return: Created node
         :rtype : :class:`Node`
         """
-        action = API_ROOT + '/instance'
+        action = '%s/cloud/project/%s/instance' % (API_ROOT, self.project_id)
         data = {
             'name': name,
             'imageId': image.id,
             'flavorId': size.id,
             'region': location.id,
         }
         if ex_keyname:
-            data['sshKeyName'] = ex_keyname
+            key_id = self.get_key_pair(ex_keyname, location).extra['id']
+            data['sshKeyId'] = key_id
         response = self.connection.request(action, data=data, method='POST')
         return self._to_node(response.object)
 
     def destroy_node(self, node):
-        action = API_ROOT + '/instance/' + node.id
+        action = '%s/cloud/project/%s/instance/%s' % (
+            API_ROOT, self.project_id, node.id)
         self.connection.request(action, method='DELETE')
         return True
 
     def list_sizes(self, location=None):
-        action = API_ROOT + '/flavor'
-        data = {}
+        action = '%s/cloud/project/%s/flavor' % (API_ROOT, self.project_id)
```

Comment:
You make this transformation a lot. I would either store this as an instance level attribute or have a private method for `_request_for_project`.